app/javascript/app/styles/layout.scss

```diff
@@ -27,12 +27,26 @@ $grid-padding-gutters: $gutter-padding;
   }
 }
 
-@mixin columns($number, $inner: false) {
+@mixin columns($number, $gutter-size: $grid-margin-gutters, $inner: false) {
   @include xy-grid($wrap: true);
-  @include xy-gutters($negative: $inner);
+  @include xy-gutters($gutters: $gutter-size, $negative: $inner);
 
   > * {
-    @include xy-cell(12/$number);
+    @include xy-cell($size: 12/$number, $gutters: $gutter-size);
+  }
+}
+
+@mixin customColumns($columns, $grid-gutters: $gutter-padding, $cell-gutters: $gutter-padding) {
```

Comment:
It would be nice to have the same name for the gutter variables in the two mixins. So have grid-gutters and cell-gutters for both for example. Also, merge the two mixins if possible as they are very similar. But this could be a bit harder.

Reply:
Agree with the variables naming.  
About merging both mixins maybe could become less explicit, in the classes using them, the kind of grid that we are trying to create. I would go with both of them.